pkg/virtctl/root.go

```diff
@@ -36,6 +36,29 @@ func NewVirtctlCommand() *cobra.Command {
 			fmt.Fprint(cmd.OutOrStderr(), cmd.UsageString())
 		},
 	}
+	completionCmd := &cobra.Command{
+		Use:   "completion",
+		Short: "Generates completion scripts for the specified shell",
+		Args:  cobra.MaximumNArgs(1),
+		Long: `To load completion run
+. <(virtctl completion (SHELL TYPE))
+
+To configure bash shell to load completions for each session add to bashrc
+
+# ~/.bashrc or ~/.profile
+. <(virtctl completion (SHELL TYPE))
+`,
+		Run: func(cmd *cobra.Command, args []string) {
+			switch {
+			case len(args) == 0:
```

Comment:
Validation should not be necessary in the command code itself. You can define it on the command above itself.